clap_complete/src/dynamic/completer.rs

```diff
@@ -389,6 +418,23 @@ fn complete_arg_value(
         values.sort();
     }
 
+    if let Some(prefix) = prefix {
+        values = values
+            .into_iter()
+            .map(|comp| {
+                CompletionCandidate::new(format!(
+                    "{}{}{}",
+                    prefix,
+                    arg.get_value_delimiter()
+                        .map(|c| c.to_string())
+                        .unwrap_or("".to_string()),
+                    comp.get_content().to_string_lossy()
+                ))
+                .help(comp.get_help().cloned())
+                .visible(comp.is_visible())
```

Comment:
Should we have a method on `CompletionCandidate` to add prefixes or to `map` the candidate?  I'm concerned that by manually recreating `CompleteCandidate`, we'll drop information as new information gets added and we forget to update this

Reply:
I also think manually rebuilding the `CompletionCandidate` is not good, especially with the possibility of more fields in the future.